extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -969,10 +969,11 @@ private StageOutputHolder getOrCreateStageOutputHolder(final StageId stageId, fi
 
   /**
    * Returns cancellation ID for a particular stage, to be used in {@link FrameProcessorExecutor#cancel(String)}.
+   * In addition to being a token for cancellation, this also appears in thread dumps, so make it a little descriptive.
    */
   private static String cancellationIdFor(final StageId stageId, final int workerNumber)
   {
-    return StringUtils.format("%s_%s", stageId, workerNumber);
+    return StringUtils.format("dart-worker[%s_%s]", stageId, workerNumber);
```

Comment:
Is this useful for non-Dart workers too?

Reply:
Oops, this should say `msq-worker`, not `dart-worker`, since this isn't Dart specific code. Updated.